extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/AbstractGuacamoleTunnelService.java

```diff
@@ -416,6 +462,21 @@ private GuacamoleTunnel assignGuacamoleTunnel(ActiveConnectionRecord activeConne
             GuacamoleClientInformation info, Map<String, String> tokens,
             boolean interceptErrors) throws GuacamoleException {
 
+        // Set up JDBC-specific tokens
+        tokens.put(JDBC_DATE_TOKEN,
+                new SimpleDateFormat(JDBC_DATE_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
+        tokens.put(JDBC_TIME_TOKEN,
+                new SimpleDateFormat(JDBC_TIME_TOKEN_FORMAT)
+                        .format(activeConnection.getStartDate()));
+        tokens.put(JDBC_CONNECTION_NAME_TOKEN, activeConnection.getConnectionName());
+        tokens.put(JDBC_CONNECTION_ID_TOKEN, activeConnection.getConnectionIdentifier());
```

Comment:
I'm not sure about adding tokens specific to JDBC for the identifier and name. What about instead implementing `GUAC_CONNECTION_NAME` / `GUAC_CONNECTION_ID` at the webapp level via `StandardTokenMap`?

And what is the intended behavior for these if a balancing connection group is used?

Reply:
Yeah, I'm fine implementing these at a higher level. I think I originally looked at it, but seemed like there were some challenges with grabbing those items for connections not stored in JDBC. I'll take another look, though.

For a balancing group, it's a good point. Seems like there are two possible options - or perhaps we should do some combination of both of these:
* Use the value of the balancing group name & id.
* Use the value of the connection that is actually selected.

Seems like having both available might be useful - there may be circumstances where it'd be useful - for example, when generating recording names - to have the name of both the balancing group and the underlying connection.

Reply:
Ah, yes, a quick look back at that reminds me why I didn't add it in `StandardTokenMap`. Currently `StandardTokenMap` is only invoked within `TunnelRequestService`, and, when it gets called there, the data provided is somewhat limited. The connection/group name is not provided, and the identifier is a bit abstract. I'm not sure how much re-tooling we want to do of that entire chain of classes just to provide these tokens, rather than just relying on the various `Connection` and `ConnectionGroup` implementations to identify them?

Reply:
@mike-jumper Are you good with the current approach, or still want to try to re-work this into the `StandardTokenMap`?